cpp/tests/prims/mg_transform_reduce_e.cu

```diff
@@ -0,0 +1,445 @@
+                                                           raft::handle_t const& handle)
+  {
+    auto length = thrust::distance(begin, end);
+    auto data   = std::make_tuple(rmm::device_uvector<Args>(length, handle.get_stream())...);
+    auto zip    = get_zip_iterator(data);
+    thrust::transform(handle.get_thrust_policy(),
+                      begin,
+                      end,
+                      zip,
+                      property_transform<label_t, Args...>(hash_bin_count));
+    return data;
+  }
+  template <typename Op, typename T1, typename T2, std::size_t... I>
+    static constexpr void copy_property_impl(Op&& op, T1&& t1, T2&& t2, std::index_sequence<I...>)
+    {
+      (op(std::get<I>(t1), std::get<I>(t2)), ...);
+    }
+
+  template <typename Op, typename Tuple>
+    static void
+    copy_property(Tuple const& property,
+                  Tuple& output_property,
+                  Op op)
+    {
+      copy_property_impl(op, property, output_property, std::make_index_sequence<std::tuple_size<Tuple>::value>());
+    }
+
+  template <typename GraphViewType>
+    static std::tuple<rmm::device_uvector<Args>...>
+    column_property(raft::handle_t const& handle,
+                    GraphViewType const& graph_view,
+                    std::tuple<rmm::device_uvector<Args>...>& property)
+    {
+      if (true) {
+        std::cerr<<"ERR DEBUG MESSAGE "<<graph_view.get_number_of_local_adj_matrix_partition_cols()<<" "<<
+          graph_view.get_number_of_local_vertices()<<"\n";
+        cudaDeviceSynchronize();
+      }
+      auto output_property = std::make_tuple(rmm::device_uvector<Args>(graph_view.get_number_of_local_adj_matrix_partition_cols(), handle.get_stream())...);
+      copy_property(property, output_property,
+                    [&handle, &graph_view] (const auto& in, auto& out) {
+                      copy_to_adj_matrix_col(handle, graph_view, in.begin(), out.begin());
+                    }
+                   );
+      return output_property;
+    }
+
+  template <typename GraphViewType>
+    static std::tuple<rmm::device_uvector<Args>...>
+    row_property(raft::handle_t const& handle,
+                    GraphViewType const& graph_view,
+                    std::tuple<rmm::device_uvector<Args>...>& property)
+    {
+      if (true) {
+        std::cerr<<"ERR DEBUG MESSAGE "<<graph_view.get_number_of_local_adj_matrix_partition_rows()<<" "<<
+          graph_view.get_number_of_local_vertices()<<"\n";
+        cudaDeviceSynchronize();
+      }
+      auto output_property = std::make_tuple(rmm::device_uvector<Args>(graph_view.get_number_of_local_adj_matrix_partition_rows(), handle.get_stream())...);
+      copy_property(property, output_property,
+                    [&handle, &graph_view] (const auto& in, auto& out) {
+                      copy_to_adj_matrix_row(handle, graph_view, in.begin(), out.begin());
+                    }
+                   );
+      return output_property;
+    }
+
+};
```

Comment:
FYI:

Actually, I am adding

```
template <typename GraphViewType, typename T>
class row(or col)_properties_t;
```
And this will be the last input parameter of copy_to_adj_matrix_row(...);

Reply:
It will be completely covered by #1801 (but its actually behavior will not change, the actual memory footprint scaling improvements will happen in another PR, but this will be transparent to primitives users).